protostar/profiler/transaction_profiler_test.py

```diff
@@ -0,0 +1,223 @@
+from dataclasses import replace
+import pytest
+from starkware.starknet.business_logic.execution.execute_entry_point import (
+    ContractEntryPoint,
+)
+from protostar.profiler.contract_profiler import Function, RuntimeProfile, Instruction
+from protostar.profiler.transaction_profiler import (
+    build_global_functions,
+    build_global_instructions,
+    get_instruction_id_offsets,
+    translate_callstack,
+)
+
+from protostar.starknet.cheatable_execute_entry_point import ContractProfile
+
+profile = RuntimeProfile(
+    [Function("f1", "d", 1), Function("f2", "d", 2)], [], [], [], []
+)
+
+
+@pytest.mark.parametrize(
+    "samples,result",
+    [
+        (
+            [
+                ContractProfile(["a", "b"], ContractEntryPoint(0, 0), replace(profile)),
```

Comment:
Why is `replace` necessary here?

Reply:
All of them use the same structure, I want to avoid them having the same reference